### PR TITLE
Cleaned Up Glue and Allowlists

### PR DESCRIPTION
**Pull request by redfire75369**
Removed Unnecessary Items in Allowlists

### CODE REVIEW

**Comment by sagudev** [comment]
I will review this and make companion PR to servo.

**Comment by sagudev** [comment]
We need to do sync anyway. Also better safe then sorry.

File: mozjs/src/conversions.rs [hunk 1]
```diff
@@ -411,7 +410,7 @@ impl FromJSValConvertible for i64 {
 impl ToJSValConvertible for u64 {
     #[inline]
     unsafe fn to_jsval(&self, _cx: *mut JSContext, mut rval: MutableHandleValue) {
-        RUST_JS_NumberValue(*self as f64, &mut *rval);
+        rval.set(DoubleValue(*self as f64));
```
[the remark this answers is not in the document]
Comment (Redfire75369) [reply mozjs/src/conversions.rs:413]: We already have tests to make sure that our functions relating to `Value` work the same way, so this should be fine.

[the remark this answers is not in the document]
Comment (sagudev) [reply mozjs/src/conversions.rs:413]: WPT tests are also happy: https://github.com/servo/servo/pull/30836

**Review by sagudev — APPROVED** [review]

**Pull request merged**